app/models/game.rb

```diff
@@ -3,26 +3,64 @@ class Game < ApplicationRecord
   belongs_to :white_player, class_name: 'User', optional: true
   has_many :users
   has_many :pieces
-
   enum game_status: { in_progress: 0, checkmate: 1, stalemate: 2 }
 
   scope :available, -> { where('black_player_id IS NULL OR white_player_id IS NULL') }
 
   def check?
-    pieces.each do |piece|
-      color = piece.color
-
-      if color == 'white'
-        other_king = pieces.find_by(type: 'King', color: 'black')
-        return true if piece.valid_move?(other_king.row_coordinate, other_king.column_coordinate)
-      elsif color == 'black'
-        other_king = pieces.find_by(type: 'King', color: 'white')
-        return true if piece.valid_move?(other_king.row_coordinate, other_king.column_coordinate)
+    color_opponent = turn == black_player_id ? 'white' : 'black'
```

Comment:
A method for getting the opponents color could be useful (that tends to get used in multiple places)